### ISSUE TITLE
SMPTE 2022-1 2D Forward Error Correction in GStreamer

### ISSUE THREAD

**Issue by MathieuDuponchelle** [MathieuDuponchelle/MathieuDuponchelle.github.io#4]

**Comment by xhejtman** [comment]
Is it usable also for webrtc from gstreamer? And for low-latency desktop streaming?

**Comment by MathieuDuponchelle** [comment]
@xhejtman no, webRTC only specifies support for flexfec and ulpfec. Regarding low-latency desktop streaming, I can't answer, this isn't a protocol :) If you were to roll your own solution using GStreamer, then you could certainly use this yes.

**Comment by MathieuDuponchelle** [comment]
@yuyou you probably need a more recent version of GStreamer

**Comment by MathieuDuponchelle** [comment]
@mrayy collect a stacktrace to the assertions, check `GST_DEBUG=2` logs

**Comment by MathieuDuponchelle** [comment]
most likely `rtpst2022-1-fecdec` isn't available, check `gst-inspect-1.0 rtpst2022-1-fecdec`

**Comment by MathieuDuponchelle** [comment]
oh I see, that's good to hear because it would have otherwise meant a regression in gst-parse-launch's grammar and debugging in there is no fun :)

**Comment by wvmcastro** [comment]
Hi @MathieuDuponchelle,

I am building an rtp relay application with rtpbin element. It is divided into one Receiver and multiple Transmits. The receiver pipeline uses rtpst2022-1-fedec element. When I use an udpsink/udpsrc pair to send data from the receiver to a transmit, things goes well. But when I use appsink/appsrc the video breaks, as if the IFrames were dropped.

The interesting bit is that when I remove fecdec from the receiver pipeline, the video comes back. Using log level 5, I could see that every IFrame recovered by the fec decoder is detected as a Jump in seqnum by a downstream rtph264depay element.

I suppose it is an interation problem between appsink and how the fecdec may (?) change the packet after correcting it. The receiver pipeline is below
```
  rtpbin name=receive rtp-profile=avpf latency=250 do-retransmission=true
         fec-decoders=fec,0="rtpst2022-1-fecdec\ size-time\=1000000000";
  udpsrc port={{video_port}}
         caps="application/x-rtp,media=(string)video,clock-rate=(int)90000,
         encoding-name=(string)H264,payload=(int)96"
    ! receive.recv_rtp_sink_0
  receive.recv_rtp_src_0_0_96
    ! appsink emit-signals=TRUE name="rtp_stream"
...
```
I am pulling data from the appsink with a callback registered for the new-sample signal. I would like any thoughts or suggestions since I run out of ideas to debug this

**Comment by MathieuDuponchelle** [comment]
@wvmcastro please file an issue @ https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues :)
